ring/lifecycler.go

```diff
@@ -795,24 +805,28 @@ func (i *Lifecycler) changeState(ctx context.Context, state InstanceState) error
 
 func (i *Lifecycler) updateCounters(ringDesc *Desc) {
 	healthyInstancesCount := 0
-	zones := map[string]struct{}{}
+	zones := map[string]int{}
 
 	if ringDesc != nil {
 		now := time.Now()
 
 		for _, ingester := range ringDesc.Ingesters {
-			zones[ingester.Zone] = struct{}{}
+			if _, ok := zones[ingester.Zone]; !ok {
+				zones[ingester.Zone] = 0
+			}
```

Comment:
Nit: Setting default value `0` to `zones[ingester.Zone]` is unnecessary. `zones[ingester.Zone]++` below will work just fine without this.

Reply:
Hi @pstibrany, 
This assignment is needed in order to count the number of zones. If we remove the block setting `0` to `zones[ingester.Zone]`, we will count only the number of zones with healthy ingesters.

Reply:
Good point that I missed! That's worth adding a comment in my opinion.